src/kloewe/dot/src/dot.h

```diff
@@ -0,0 +1,92 @@
+/*----------------------------------------------------------------------------
+  File    : dot.h
+  Contents: dot product (cpu dispatcher)
+  Author  : Kristian Loewe
+----------------------------------------------------------------------------*/
+#ifndef DOT_H
+#define DOT_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/*----------------------------------------------------------------------------
+  Enum to encode the sets of dot implementations
+----------------------------------------------------------------------------*/
+enum flags {
+    DOT_NAIVE  = 1,
+    DOT_SSE2   = 2,
+    DOT_AVX    = 3,
+    #ifndef DOT_NOFMA
+    DOT_AVXFMA = 4,
+    #endif
+    DOT_AUTO   = 3
```

Comment:
That's interesting. Some time ago, I did try to manually unroll the loop body by 2. In my small test program, that seemed to improve the avx-fma variants a bit. Strangely though, when I used this in -cifti-correlation and -cifti-correlation-gradient the manually unrolled version was not faster than the regular one. I'll have to look into this further at some point.